test/inductor/test_max_autotune.py

```diff
@@ -463,6 +467,78 @@ def addmm(x, a, b):
         # given the config flags above, we should have no choices left.
         self.assertIn("NoValidChoicesError", str(context.exception))
 
+    @unittest.skipIf(TEST_WITH_ROCM, "ROCm doesn't support sm carveout")
+    @unittest.skipIf(IS_WINDOWS, "Windows doesn't support persistent TMA")
+    @unittest.skipIf(
+        not has_triton_tma_device(), "Need device-side TMA support in Triton"
+    )
+    def test_honor_sm_carveout_with_triton_tma(self):
+        def mm_func(a, b):
+            return torch.mm(a, b)
+
+        # Create large matrices to ensure we use all possible sms
+        size = 2560
+        a = torch.randn(size, size, device="cuda", dtype=torch.bfloat16)
+        b = torch.randn(size, size, device="cuda", dtype=torch.bfloat16)
+
+        with config.patch(
+            {
+                "max_autotune": True,
+                "autotune_fallback_to_aten": False,
+                "triton.enable_persistent_tma_matmul": True,
+                "max_autotune_gemm_backends": "TRITON",
+                "test_configs.autotune_choice_name_regex": "mm_persistent_tma",
+            }
+        ):
+            compiled_mm = torch.compile(mm_func, mode="max-autotune-no-cudagraphs")
+            compiled_mm(a, b)
+
+            with tempfile.NamedTemporaryFile() as f:
+                with torch.profiler.profile(
+                    activities=[torch.profiler.ProfilerActivity.CUDA]
+                ) as prof:
+                    # Test with default (no carveout)
+                    self.assertIsNone(torch._C._get_sm_carveout_experimental())
+                    compiled_mm(a, b)
+
+                    # Test with carveout 0
+                    torch._C._set_sm_carveout_experimental(0)
+                    self.assertEqual(torch._C._get_sm_carveout_experimental(), 0)
+                    compiled_mm(a, b)
+
+                    # Test with carveout 66
```

Comment:
In my previous tests for fp8 I chose 66 because the tests only ran on H100 which has 132, so I thought that picking half of that would have the highest chances of giving a visible difference.